### ISSUE TITLE
How to detect the edge of our own data

### ISSUE THREAD

**Issue by xwhkkk** [MengyangPu/EDTER#81]
Thank you for sharing your great job !
How can we detect the edge of our own jpg or png files? Can we directly use the pre-trained model?

Thank you for your suggestions!

**Comment by MengyangPu** [comment]
Hi, you could try to use the pre-trained model to detect your images. 
Firstly, prepare a list of images according to the file(test.txt)[data/BSDS/ImageSets/test.txt] of BSDS. The first column is the path of your image. The second column can be set to a path ending with .mat, which is only used for placeholders.
Second, Refer to the (tutorial)[https://github.com/MengyangPu/EDTER#3-testing] for detection.

**Comment by Arslan-Mehmood1** [comment]
> Hi, you could try to use the pre-trained model to detect your images. Firstly, prepare a list of images according to the file [test.txt](https://github.com/MengyangPu/EDTER/tree/main/data/BSDS/ImageSets/test.txt) of BSDS. The first column is the path of your image. The second column can be set to a path ending with .mat, which is only used for placeholders. Second, Refer to the [tutorial](https://github.com/MengyangPu/EDTER#3-testing) for detection.

and where do we need to put the path of test.txt?
The process to make inference is not clear, in tutorial there is no guidance on setting the custom test images!
isn't there any way of using the pretrained checkpoints to make direct predicitons instead of using this long process of test.py?

**Comment by MengyangPu** [comment]
@BugMaker-bot 
Currently, reading data consisting solely of PNG files is not supported. However, please rest assured that MAT files are not being used. Refer to the [test.txt](https://github.com/MengyangPu/EDTER/blob/main/data/BSDS/ImageSets/test.txt) file to create your data list.
If you want to attempt modifying the data loading function, you can refer to [here](https://github.com/MengyangPu/EDTER/blob/3fe76f3d938206ef9dc8b857a9767b8cd3d28fc7/mmseg/datasets/pipelines/loading.py#L95)

**Comment by MengyangPu** [comment]
@Arslan-Mehmood1 
Refer to the [test.txt](https://github.com/MengyangPu/EDTER/blob/main/data/BSDS/ImageSets/test.txt) file to create your data list.

**Comment by MengyangPu** [comment]
@Arslan-Mehmood1 
Firstly, you can place your data wherever you find suitable, for example, /EDTER/data.
Second, create a test.txt file based on the absolute path of your data. The first column is the path of your image. The second column can be set to a path ending with .mat.(please rest assured that MAT files are not being used.)
Finally, refer to [configs/_base_/datasets/bsds.py](https://github.com/MengyangPu/EDTER/blob/main/configs/_base_/datasets/bsds.py) to create your custom data configuration file.

**Comment by MengyangPu** [comment]
@Arslan-Mehmood1 
Yes, you can use the pretrained checkpoints for direct predictions. Instead of using the test.py script, you can load the pretrained model checkpoint in the PyTorch framework and then use the model to make predictions on new data. But this may require you to customize the test script.
The current test script is modified based on [mmsegmentation](https://github.com/open-mmlab/mmsegmentation). You can refer to the [original script](https://github.com/open-mmlab/mmsegmentation/blob/main/tools/test.py) file for guidance.

**Comment by Arslan-Mehmood1** [comment]
thanks @MengyangPu  for such a detailed response.
I'll work on it and share my feedback

best regards,
Arslan

**Comment by 2783000** [comment]
> thanks @MengyangPu for such a detailed response. I'll work on it and share my feedback
> 
> best regards, Arslan

Could you share how you did it?

**Comment by 7zburger** [comment]
Can you edge-eval on your own dataset?

**Comment by grzhao89** [comment]
I tested on my own dataset, stage I worked well but stage II didn't. It kept showing error that 2 tensor dimensions are different on fuse head part. I don't want to resize my images, so I got stuck there.

**Comment by Gao-yudong** [comment]
> I tested on my own dataset, stage I worked well but stage II didn't. It kept showing error that 2 tensor dimensions are different on fuse head part. I don't want to resize my images, so I got stuck there.
Hi, I encountered the same issue. Have you resolved it?

**Comment by grzhao89** [comment]
> > I tested on my own dataset, stage I worked well but stage II didn't. It kept showing error that 2 tensor dimensions are different on fuse head part. I don't want to resize my images, so I got stuck there.
> > Hi, I encountered the same issue. Have you resolved it?

It is resolved. You can try to debug the file segmentors/encoder_decoder_local8x8.py, and I modified the "def aug_test" part which you need to adjust the dimensions there.
